pkg/clientcmd/configutil.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clientcmd
+
+import (
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/clientcmd/api"
+	"sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset"
+)
+
+// This file contains convenience methods for working with the kubeconfig and loading specific configurations from it
+
+// Creates an api.Config by searching for the kubeconfig on the default search path. If an override 'kubeconfigPath' is
+// given then that path is used instead of the default path.
+func NewApiConfigForDefaultSearchPath(kubeconfigPath string) (*api.Config, error) {
+	configLoader := clientcmd.NewDefaultClientConfigLoadingRules()
+	configLoader.ExplicitPath = kubeconfigPath
```

Comment:
Does this work correctly when the path is empty? Or should it be wrapped in an if check?

Reply:
Yes it does. ConfigLoader supports having no ExplicitPath in which case it uses the standard load order (${KUBECONFIG}, ${HOME}/.kube/config etc). Note that explicit path is an actual 'string' and not a pointer value, so effectively setting it to an empty string has no effect because that was the value it already had.

Reply:
This pattern is weird though, because it's possible that `NewDefaultClientConfigLoadingRules()` would change `ExplicitPath` from the default value to something useful and this overrides it back to the default value. If that behavior were changed in the future, then this would turn into a bug, whereas if you passed kubeconfigPath into the loading rules fn it could take it or leave it (or if we had a builder). 

This certainly isn't your problem to solve, it's just confusing for reviewers.